### PR TITLE
Fix validation progress counter with check_val_every_n_epoch > 1

### LINKED ISSUE #5039: validating bar keep showing 0 when check_val_every_n_epoch is bigger than 1
## 🐛 Bug

Normally, the validating bar should update according to the procedure of validation_step, but when I set check_val_every_n_epoch bigger than 1, 2 for example, it keeps showing 0it.

I encountered this issue under a terminal environment.

 - PyTorch 1.6.0:
 - Linux

### PR DESCRIPTION
**Pull request by sailordiary**
## What does this PR do?

Fixes #5039

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [ ] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by carmocca** [comment]
Hi. Thanks for your contribution!

Can you write a test so we avoid regressions?

**Comment by sailordiary** [comment]
> Hi. Thanks for your contribution!
> 
> Can you write a test so we avoid regressions?

Happy to!  But I just stumbled upon #5413 and it seems that there was a consesus to deprecate the `check_val_every_n_epoch` argument soon, so I'm not sure if this patch would still be necessary...

**Comment by tchaton** [comment]
Hey @sailordiary,

Any updates on this PR ? It seems there were a discussion on the other issue, but no PR were associated to it.

Best,
T.C

**Comment by rohitgr7** [comment]
@sailordiary you can go ahead and complete this PR with a test for now. Since this is a bug and even if we are going to remove this in the future it will be deprecated first. So bugfix is required here. Mind add a test for it?

**Comment by tchaton** [comment]
Hey @rohitgr7,

Mind adding a test to this PR, so we can merge it ?

Best,
T.C

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ default_language_version:
 
 repos:
   - repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v2.3.0
+    rev: v3.4.0
```
Comment (rohitgr7) [inline .pre-commit-config.yaml:20]:
```suggestion
    rev: v2.3.0
```

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -35,6 +35,6 @@ repos:
         args: [--parallel, --in-place]
 
   - repo: https://github.com/pre-commit/mirrors-mypy
-    rev: v0.790
+    rev: v0.812
```
Comment (rohitgr7) [inline .pre-commit-config.yaml:38]:
```suggestion
    rev: v0.790
```

**Review by rohitgr7 — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

File: tests/trainer/flags/test_check_val_every_n_epoch.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    ]
+)
+def test_check_val_every_n_epoch(tmpdir, max_epochs, expected_val_loop_calls, expected_val_batches):
+
+    class TestModel(BoringModel):
+
+        def __init__(self):
+            super().__init__()
+            self.val_epoch_calls = 0
+            self.val_batches = []
```
Comment (carmocca) [inline tests/trainer/flags/test_check_val_every_n_epoch.py:35]:
```suggestion
        val_epoch_calls = 0
        val_batches = []
```

**Review by akihironitta — APPROVED** [review]

**Pull request merged**